tests/test_workers/test_scm.py

```diff
@@ -116,11 +120,27 @@ def test_fetch_source_by_pull(mock_update_and_archive, mock_getctime):
     po = mock.patch.object
     with po(scm_git.sources_dir.archive_path, "exists", return_value=False):
         with po(
-            scm_git.sources_dir.package_dir, "glob", return_value=["29eh2a.tar.gz", "a8c2d2.tar.gz"]
+            scm_git.sources_dir.package_dir,
+            "glob",
+            return_value=["a8c2d2.tar.gz", "a8c2d2-with-submodules.tar.gz"],
```

Comment:
Should we use 3 different archives here?
1. no submodules, oldest
2. no submodules, second oldest
3. with submodules, newest

Archive 2 should be used (to test that newest goes first but `-with-submodules` is ignored)